protocols/request-response/src/lib.rs

```diff
@@ -515,33 +549,61 @@ where
             ConnectedPoint::Dialer { address } => Some(address.clone()),
             ConnectedPoint::Listener { .. } => None
         };
-        let connections = self.connected.entry(peer.clone()).or_default();
-        connections.push(Connection { id: *conn, address })
+        self.connected.entry(peer.clone())
+            .or_default()
+            .push(Connection::new(*conn, address));
     }
 
-    fn inject_connection_closed(&mut self, peer: &PeerId, conn: &ConnectionId, _: &ConnectedPoint) {
-        if let Some(connections) = self.connected.get_mut(peer) {
-            if let Some(pos) = connections.iter().position(|c| &c.id == conn) {
-                connections.remove(pos);
+    fn inject_connection_closed(&mut self, peer_id: &PeerId, conn: &ConnectionId, _: &ConnectedPoint) {
+        let connections = match self.connected.get_mut(peer_id) {
+            Some(peer) => peer,
+            None => {
+                debug_assert!(
+                    false,
+                    "Expected some established connection to peer before closing.",
+                );
+                return
```

Comment:
For the sake of simplicity, I think I would just prefer an `.expect(...)` then, which also panics in release mode.

Reply:
:+1: for the sake of consistency I will replace the `debug_assert` both for expecting to be connected to the peer and expecting an established connection for the given connection id right below it.